tests/storages_tests/test_storages.py

```diff
@@ -352,336 +452,406 @@ def test_get_trial_number_from_id(storage_mode):
+        check_set_and_get(trial_id_1, "dataset", "ImageNet")
 
-    # Test overwriting value.
-    check_set_and_get(trial_id_1, "dataset", "ImageNet")
+        # Test another trial.
+        trial_id_2 = storage.create_new_trial(storage.create_new_study())
+        check_set_and_get(trial_id_2, "baseline_score", 0.001)
+        assert len(storage.get_trial(trial_id_2).user_attrs) == 1
+        assert storage.get_trial(trial_id_2).user_attrs["baseline_score"] == 0.001
 
-    # Test another trial.
-    trial_id_2 = storage.create_new_trial(study_id)
-    check_set_and_get(trial_id_2, "baseline_score", 0.001)
-    system_attrs = storage.get_trial(trial_id_2).system_attrs
-    assert system_attrs == {"baseline_score": 0.001}
+        # Cannot set attributes of non-existent trials.
+        non_existent_trial_id = max({trial_id_1, trial_id_2}) + 1
+        with pytest.raises(KeyError):
+            storage.set_trial_user_attr(non_existent_trial_id, "key", "value")
 
+        # Cannot set attributes of finished trials.
+        storage.set_trial_state(trial_id_1, TrialState.COMPLETE)
+        with pytest.raises(RuntimeError):
+            storage.set_trial_user_attr(trial_id_1, "key", "value")
 
-@parametrize_storage
-def test_get_all_study_summaries(storage_init_func):
-    # type: (Callable[[], BaseStorage]) -> None
 
-    storage = storage_init_func()
-    study_id = storage.create_new_study()
+@pytest.mark.parametrize("storage_mode", STORAGE_MODES)
+def test_get_system_user_attrs(storage_mode: str) -> None:
 
-    storage.set_study_direction(study_id, StudyDirection.MINIMIZE)
+    with StorageSupplier(storage_mode) as storage:
+        _, study_to_trials = _setup_studies(storage, n_study=2, n_trial=5, seed=10)
+        assert all(
+            storage.get_trial_system_attrs(trial_id) == trial.system_attrs
+            for trials in study_to_trials.values()
+            for trial_id, trial in trials.items()
+        )
 
-    datetime_1 = datetime.now()
+        non_existent_trial = max(tid for ts in study_to_trials.values() for tid in ts.keys()) + 1
+        with pytest.raises(KeyError):
+            storage.get_trial_system_attrs(non_existent_trial)
 
-    # Set up trial 1.
-    _create_new_trial_with_example_trial(
-        storage, study_id, EXAMPLE_DISTRIBUTIONS, EXAMPLE_TRIALS[0]
-    )
 
-    datetime_2 = datetime.now()
+@pytest.mark.parametrize("storage_mode", STORAGE_MODES)
+def test_set_trial_system_attr(storage_mode: str) -> None:
 
-    # Set up trial 2.
-    trial_id_2 = storage.create_new_trial(study_id)
-    storage.set_trial_value(trial_id_2, 2.0)
+    with StorageSupplier(storage_mode) as storage:
+        study_id = storage.create_new_study()
+        trial_id_1 = storage.create_new_trial(study_id)
 
-    for key, value in EXAMPLE_ATTRS.items():
-        storage.set_study_user_attr(study_id, key, value)
+        def check_set_and_get(trial_id, key, value):
+            # type: (int, str, Any) -> None
```

Comment:
Please use Python3-style type hinting if you do not have specific reasons.
```suggestion
        def check_set_and_get(trial_id: int, key: str, value: Any) -> None:
```